lightning/src/util/persist.rs

```diff
@@ -243,10 +250,886 @@ where
+
+		let chain_mon_0 = test_utils::TestChainMonitor::new(
+			Some(&chanmon_cfgs[0].chain_source),
+			&chanmon_cfgs[0].tx_broadcaster,
+			&chanmon_cfgs[0].logger,
+			&chanmon_cfgs[0].fee_estimator,
+			&persister_0,
+			node_cfgs[0].keys_manager,
+		);
+		let chain_mon_1 = test_utils::TestChainMonitor::new(
+			Some(&chanmon_cfgs[1].chain_source),
+			&chanmon_cfgs[1].tx_broadcaster,
+			&chanmon_cfgs[1].logger,
+			&chanmon_cfgs[1].fee_estimator,
+			&persister_1,
+			node_cfgs[1].keys_manager,
+		);
+		node_cfgs[0].chain_monitor = chain_mon_0;
+		node_cfgs[1].chain_monitor = chain_mon_1;
+		let node_chanmgrs = create_node_chanmgrs(2, &node_cfgs, &[None, None]);
+		let nodes = create_network(2, &node_cfgs, &node_chanmgrs);
+
+		let broadcaster_0 = &chanmon_cfgs[2].tx_broadcaster;
+		let broadcaster_1 = &chanmon_cfgs[3].tx_broadcaster;
+
+		// Check that the persisted channel data is empty before any channels are
+		// open.
+		let mut persisted_chan_data_0 = persister_0
+			.read_channel_monitors_with_updates(&broadcaster_0, &chanmon_cfgs[0].fee_estimator)
+			.unwrap();
+		assert_eq!(persisted_chan_data_0.len(), 0);
+		let mut persisted_chan_data_1 = persister_1
+			.read_channel_monitors_with_updates(&broadcaster_1, &chanmon_cfgs[1].fee_estimator)
+			.unwrap();
+		assert_eq!(persisted_chan_data_1.len(), 0);
+
+		// Helper to make sure the channel is on the expected update ID.
+		macro_rules! check_persisted_data {
+			($expected_update_id: expr) => {
+				persisted_chan_data_0 = persister_0
+					.read_channel_monitors_with_updates(
+						&broadcaster_0,
+						&chanmon_cfgs[0].fee_estimator,
+					)
+					.unwrap();
+				assert_eq!(persisted_chan_data_0.len(), 1);
+				for (_, mon) in persisted_chan_data_0.iter() {
+					assert_eq!(mon.get_latest_update_id(), $expected_update_id);
+				}
+				persisted_chan_data_1 = persister_1
+					.read_channel_monitors_with_updates(
+						&broadcaster_1,
+						&chanmon_cfgs[1].fee_estimator,
+					)
+					.unwrap();
+				assert_eq!(persisted_chan_data_1.len(), 1);
+				for (_, mon) in persisted_chan_data_1.iter() {
+					assert_eq!(mon.get_latest_update_id(), $expected_update_id);
+				}
+			};
+		}
+
+		// Create some initial channel and check that a channel was persisted.
+		let _ = create_announced_chan_between_nodes(&nodes, 0, 1);
+		check_persisted_data!(0);
+
+		// Send a few payments and make sure the monitors are updated to the latest.
+		send_payment(&nodes[0], &vec![&nodes[1]][..], 8_000_000);
+		check_persisted_data!(EXPECTED_UPDATES_PER_PAYMENT);
+		send_payment(&nodes[1], &vec![&nodes[0]][..], 4_000_000);
+		check_persisted_data!(2 * EXPECTED_UPDATES_PER_PAYMENT);
+
+		// Send a few more payments.
+		let mut sender = 0;
+		for i in 1..=100 {
+			let receiver;
+			if sender == 0 { sender = 1; receiver = 0; } else { sender = 0; receiver = 1; }
+			send_payment(&nodes[sender], &vec![&nodes[receiver]][..], 21_000);
+			check_persisted_data!(10 + i * EXPECTED_UPDATES_PER_PAYMENT);
```

Comment:
Could we start the counter at 3 instead of using the magic number 10?

Reply:
Thanks... gotta laugh at myself here, I remember adding the constant so I could get this loop to work, and slapping the magic number in there and thinking "this is brittle," and then I'm pretty sure I got called AFK and forgot all about it